build error: result type method call is not checked.

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.3.1 f922ed0** 
**OS: ArchLinux** 

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
struct Foo{}

fn (f Foo) foo() int{
	return 1
}

fn new_foo() !Foo{
	return Foo{}
}

fn main(){
	i := new_foo().foo()
}
```

**What did you expect to see?**
Error message
 
**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 00000504
tcc: error: file 'crt1.o' not found
/tmp/v_1000/bug.9449201155251194160.tmp.c:12093: error: cannot convert 'struct _result_main__Foo' to 'struct main__Foo'
...
==================
```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1199,6 +1199,9 @@ pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 	if left_type.has_flag(.optional) {
 		c.error('optional type cannot be called directly', node.left.pos())
 		return ast.void_type
+	} else if left_type.has_flag(.result) {
+		c.error('result type cannot be called directly', node.left.pos())
+		return ast.void_type
 	}
 	if left_sym.kind in [.sum_type, .interface_] {
 		if method_name == 'type_name' {

--- a/vlib/v/checker/tests/result_type_call_err.out
+++ b/vlib/v/checker/tests/result_type_call_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/result_type_call_err.vv:12:2: error: result type cannot be called directly
+   10 |
+   11 | fn main() {
+   12 |     new_foo().foo()
+      |     ~~~~~~~~~
+   13 | }

--- a/vlib/v/checker/tests/result_type_call_err.vv
+++ b/vlib/v/checker/tests/result_type_call_err.vv
@@ -1,0 +1,13 @@
+struct Foo {}
+
+fn (f Foo) foo() int {
+	return 1
+}
+
+fn new_foo() !Foo {
+	return Foo{}
+}
+
+fn main() {
+	new_foo().foo()
+}

